packages/core/src/stateUtils.ts

```diff
@@ -259,15 +259,49 @@ export const isStateId = (str: string) => str[0] === STATE_IDENTIFIER;
 
 export function getCandidates<TEvent extends EventObject>(
   stateNode: StateNode<any, TEvent>,
-  eventName: TEvent['type'] | NullEvent['type'] | '*'
+  eventName: TEvent['type'] | NullEvent['type'],
+  /**
+   * If `true`, will use SCXML event token matching semantics
+   */
+  matchTokens: boolean = false
```

Comment:
General q: is it even worth supporting this? I mean - my bet is that this will only ever be used in our tests and we could deal with that differently. For example by forking those tests that require all event descriptors to be treated as inexact.

I don't have a super-strong desire to remove this - but usually, I try to advocate for not including things like this.

Reply:
It's not public API and this is the way I solved dealing with it in our tests.

I'd rather not change the tests and make matching tokens opt-in (internal) behavior than forking the tests.

It'll also be important for anyone using XState as an SCXML processor in the future.

Reply:
could we rename this option to something else even if it's just `scxml`? even for `matchTokens === false` we are still matching tokens which makes it confusing (only the precondition for matching is different)

Reply:
> even for `matchTokens === false` we are still matching tokens

We're not matching tokens; we're either matching string equivalence (`transition.eventType === eventName`) or any string with a wildcard.

Reply:
Changed this to `partialMatch`